http_general/test_block_action.py

```diff
@@ -159,13 +198,76 @@ def test_block_action_attack_reply_not_on_req_rcv_event(self):
             expected_status_code="200",
         )
 
-        client.send_request(
+        client.make_request(
             request=f"GET / HTTP/1.1\r\nHost: frang.com\r\n\r\n",
-            expected_status_code="403",
         )
+        self.check_last_error_response(client, expected_status_code="403")
+
+        self.assertTrue(client.wait_for_connection_close())
+        self.check_fin_no_rst_in_sniffer(sniffer)
+
+
+class BlockActionReplyWithCustomErrorPage(BlockActionBase):
+    tempesta_tmpl = """
+        listen 80;
+        srv_group default {
+            server ${server_ip}:8000;
+        }
+        vhost good {
+            proxy_pass default;
+        }
+        vhost frang {
+            frang_limits {
+                http_methods GET;
+                http_resp_code_block 200 1 10;
+            }
+            proxy_pass default;
+        }
+        
+        block_action attack %s;
+        block_action error %s;
+        response_body 4* %s;
+
+        http_chain {
+            host == "bad.com"   -> block;
+            host == "good.com"  -> good;
+            host == "frang.com" -> frang;
+        }
+    """
+
+    ERROR_RESPONSE_BODY = "a" * 1000
```

Comment:
Why this particular data? If we want random data, why don’t we use automatic generation of different data? After all, the task of tests, among other things, is to find edge values and break the code.

Reply:
These tests check order in which TCP frames are sent. First - TCP PSH with response, second - TCP FIN. We should not check data type in response body in these tests, because these are tests for `block_action` directive.